epam-star-wars-heroes-wiki/epam-star-wars-heroes-wiki/UI/Search/ViewController.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  ViewController.swift
+//  epam-star-wars-heroes-wiki
+//
+//  Created by Артем Маков on 15/02/2020.
+//  Copyright © 2020 Artem Makov. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController, UISearchBarDelegate {
+    
+    @IBOutlet weak var navItem: UINavigationItem!
+    @IBOutlet weak var tableView: UITableView!
+    let searchController = UISearchController(searchResultsController: nil)
+    var heroArray:[Hero] = []
+    var filteredHeroes: [Hero] = []
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        generateData()
+        registerTableView()
+        registerSearchController()
+    }
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        let titleView = UILabel(frame: CGRect(x:0, y:0, width: 40,height: 40))
```

Comment:
Please avoid initializing UI Elements in viewWillAppear. This code executes every time, when screen appears. It's unnecessary to re-set static navItem.titleView every time. Consider to move this code in viewDidLoad